app/controllers/users.js

```diff
@@ -0,0 +1,28 @@
+const { createUser: signUpService, getUserByEmail } = require('../services/users');
+const { signUp: mapperUser } = require('../mappers/users');
+const { signUp: serializerUser } = require('../serializers/users');
+const { defaultError, badRequestError } = require('../errors');
+const logger = require('../logger');
+
+exports.signUp = async (req, res, next) => {
+  try {
+    const userData = mapperUser(req.body);
+
+    const existUser = await getUserByEmail(userData.email);
+    if (existUser) {
+      return next(badRequestError('Email already exists'));
+    }
+
+    const signUpResult = await signUpService(userData);
+
+    if (!signUpResult) {
```

Comment:
In which cases would this happen? since you are already checking for the email's existence and in the service function you are already throwing an error.